crates/oxc_semantic/src/scope.rs

```diff
@@ -19,7 +19,12 @@ type UnresolvedReferences = FxHashMap<Atom, Vec<ReferenceId>>;
 /// `SoA` (Struct of Arrays) for memory efficiency.
 #[derive(Debug, Default)]
 pub struct ScopeTree {
+    /// Maps a scope to the parent scope it belongs in
     parent_ids: IndexVec<ScopeId, Option<ScopeId>>,
+
+    /// Maps a scope to direct children scopes
+    child_ids: IndexMap<ScopeId, Vec<ScopeId>>,
```

Comment:
Should be a bit more performant with fxhasher, https://docs.rs/indexmap/latest/indexmap/#alternate-hashers